p2p/host/basic/basic_host_test.go

```diff
@@ -209,18 +208,20 @@ func TestLocalIPChangesWhenListenAddrChanges(t *testing.T) {
 	h := New(swarmt.GenSwarm(t, ctx, swarmt.OptDialOnly))
 	defer h.Close()
 
-	h.lipMu.Lock()
-	h.localIPv4Addr = nil
-	h.lipMu.Unlock()
+	h.addrMu.Lock()
+	h.filteredInterfaceAddrs = nil
+	h.allInterfaceAddrs = nil
+	h.addrMu.Unlock()
 
 	// change listen addrs and veify local IP addr is not nil again
 	require.NoError(t, h.Network().Listen(ma.StringCast("/ip4/0.0.0.0/tcp/0")))
 	h.SignalAddressChange()
 	time.Sleep(1 * time.Second)
 
-	h.lipMu.RLock()
-	h.lipMu.RUnlock()
-	require.NotNil(t, h.localIPv4Addr)
+	h.addrMu.RLock()
+	defer h.addrMu.RUnlock()
+	require.NotEmpty(t, h.filteredInterfaceAddrs)
+	require.NotEmpty(t, h.allInterfaceAddrs)
 }
 
 func TestAllAddrs(t *testing.T) {
```

Comment:
Not strictly related to this PR, but just for curiousity/education. These tests seem to focus on ipv4, do we/should we have ipv6 (and ipv4+ipv6) related tests here as well?

Reply:
Probably. I think we just picked IP4 because most IPv4/IPv6 cases should be symmetric.